sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala

```diff
@@ -1044,7 +1057,8 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer {
           // iterator has exhausted, we should clean up
           close()
         }
-        queue.nonEmpty || iter.hasNext
+        // If there are no cachedAttributes, there is nothing to read
+        !cachedAttributes.isEmpty && (queue.nonEmpty || iter.hasNext)
```

Comment:
Again this is incorrect. If there are no attributes (columns), there are still rows.  We need to be able to return a ColumnarBatch with no columns, just rows, just like the data that came in.